gemrb/core/System/MemoryStream.h

```diff
@@ -29,16 +29,16 @@ namespace GemRB {
 
 class GEM_EXPORT MemoryStream : public DataStream
 {
-private:
+protected:
 	char *data;
 public:
-	MemoryStream(char *name, void* data, unsigned long size);
-	~MemoryStream();
-	DataStream* Clone();
+	MemoryStream(const char *name, void* data, unsigned long size);
+	virtual ~MemoryStream();
+	virtual DataStream* Clone();
 
-	int Read(void* dest, unsigned int length);
-	int Write(const void* src, unsigned int length);
-	int Seek(int pos, int startpos);
+	virtual int Read(void* dest, unsigned int length);
```

Comment:
all these are already virtual, but you should mark them as `override` instead.

Reply:
I'll have a look later today.

Reply:
Ok, I learned that there is no need to manually re-virtual down the chain.

Reply:
Yup. All the `virtual` keyword does is tell the compiler to create a vtable for the function. Once a vtable exists it can't not exist. Do mark them as `override` tho.

Reply:
I fixed that. I just had the concept in mind that C++ may allow to restrict vtable entries. e. g. virt. A > B > C would lead to invoke only B even if we have C under a polymorph A when not doing virt. A > virt. B > C.